nx_cugraph/algorithms/lowest_common_ancestors.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cupy as cp
+import networkx as nx
+import numpy as np
+import pylibcugraph as plc
+
+from nx_cugraph.convert import _to_directed_graph
+from nx_cugraph.utils import (
+    _groupby,
+    index_dtype,
+    networkx_algorithm,
+    not_implemented_for,
+)
+
+__all__ = ["lowest_common_ancestor"]
+
+
+@not_implemented_for("undirected")
```

Comment:
Does adding this make the algorithm automatically raise errors on undirected graphs?

Reply:
Yes. It's a decorator from networkx.

We place it before `@networkx_algorithm`, because NetworkX already checks and raises if the input graph is undirected before dispatching to backends:
https://github.com/networkx/networkx/blob/9beaf7a0b59fe21775cd93862d9c7b28152a2d8c/networkx/algorithms/lowest_common_ancestors.py#L115-L117

In other words, we use the same decorator so the algorithm behaves correctly when used directly such as `nxcg.lowest_common_ancestor(G)`.